java/src/main/java/ai/rapids/cudf/DataType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.rapids.cudf;
+
+import java.io.IOException;
+
+public class DataType {
+
+  DType typeId;
+  int scale;
+   public DataType(DType id) {typeId = id; }
+
+   public DataType(DType id,  int fp_scale) {
+     typeId =id;
+     scale = fp_scale;
+   }
+  private static native long makeNativeType(int typeId, int scale);
+
+  public class NativeDataType implements AutoCloseable {
```

Comment:
Thanks for pointing this out. I was contemplating on this. 
Right now, we don't have any use-case, we could pass DType ordinal and scale(as I have done in current implementation) but if we need to add another field such as `precision` in DataType later, it would be easier to add.  
I will go ahead with current approach for now i.e pass ordinal and the scale.

Reply:
Hi @jlowe , as previous discussion between @nartal1 and I, we decide to introduce the class `NativeDataType`, which holds/manages the handle of native class instance `data_type`, because it can pass the handle to JNI method instead of typeID, scale and possible additional parameters in future. In this approach, we can avoid interface change for any changes on `DataType`.

class `NativeDataType` needs to be `AutoClosable` in case of memory leak in native level.  It should be initialized with try resource clause.

Reply:
Hi @sperlingxx, Thanks for reviewing. Had a discussion with @jlowe on this today morning. It is correct that we can avoid interface change for any changes on DataType in the future but it adds complexity of tracking the lifetimes(which may be prone to bugs if not closed properly).  As of now we are passing one more arg i.e `scale` in the JNI api's which is okay. As of now, there doesn't seem to be much added in `data_type` itseems and we can revisit this approach in the future if it becomes absolutely necessary.

Reply:
Okay, I believe you are right.